src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -90,33 +103,112 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         }
       }
     }
-    // TODO: apply sort param choices
-
     return results;
   }
 
   // returns a list of all events created by a user (identified by email id)
-  private List<Entity> handleCreated(String userEmail) {
+  private List<Entity> getHandleCreated(String userEmail) {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     List<Entity> results = new ArrayList<>();
 
     Query query =
         new Query("Event")
-            .setFilter(new Query.FilterPredicate("creator", Query.FilterOperator.EQUAL, userEmail))
-            .addSort("eventName", SortDirection.ASCENDING);
+            .setFilter(new Query.FilterPredicate("creator", Query.FilterOperator.EQUAL, userEmail));
+    // .addSort("eventName", SortDirection.ASCENDING);
     PreparedQuery queried = datastore.prepare(query);
     for (Entity e : queried.asIterable()) {
       results.add(e);
     }
-    // TODO: apply sort param choices
-
     return results;
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // adds or removes events from user's saved events list
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
 
-    // TODO: add event to a list
+    if (userService.isUserLoggedIn()) {
```

Comment:
Always prefer less nested logic for more readability. For instance,
```
if (!userService.isUserLoggedIn()) {
  throw new IOException("must be logged in");
}
if (request.getParameter("event") != null) {
  throw new IOException("no event key specified");
}
// Handle the logic
```

Reply:
Fixed, thanks!